src/styles/main.scss

```diff
@@ -29,72 +45,180 @@ body {
     }
 
-    &.selected {
-      background: $url-blue;
+    .result--title {
+      @include overflow-at(65%);
+    }
 
-      .result-copy {
-        color: $white;
-      }
+    .result--url {
+      @include overflow-at(30%);
+      color: $color--blue;
+    }
 
-      .result-url {
-        color: $white;
-        overflow-x: hidden;
+    .result--action {
+      @include overflow-at(30%);
+      color: $color--green;
+    }
+
+    &:hover {
+      background: $color--light-grey;
+      border-color: darken($color--light-grey, 3%);
+    }
+
+    &.result--selected {
+      background: $color--blue;
+      border-color: darken($color--blue, 3%);
+
+      .result--title,
+      .result--action,
+      .result--url,
+      .result--segment + .result--segment:before,
+      .little-machine--segment {
+        color: $color--white;
       }
 
       .icon-search-remote {
-        background-position: 0 (-1 * $icon-unit);
+        background-position: 0 (-1 * $size--icon);
       }
 
       .icon-bookmark {
-        background-position: (-1 * $icon-unit) (-1 * $icon-unit);
+        background-position: (-1 * $size--icon) (-1 * $size--icon);
       }
 
       .icon-search-local {
-        background-position: (-2 * $icon-unit) (-1 * $icon-unit);
+        background-position: (-2 * $size--icon) (-1 * $size--icon);
       }
     }
   }
+
+  .result--segment--little-machine {
+    flex-direction: column;
+    min-height: $size--preview;
+    padding-left: ($size--preview + $size--preview--spacing);
+    position: relative;
+
+    .little-machine--segment {
+      margin: 0;
+    }
+
+    .little-machine--image {
+      border-radius: 5px;
+      box-shadow: 0 0 1px rgba(0, 0, 0, 0.25) inset;
+      left: 0;
+      line-height: 0;
+      overflow: hidden;
+      position: absolute;
+      top: 0;
+
+      img {
+        height: $size--preview;
+        position: relative;
+        width: $size--preview;
+        z-index: -1;
+      }
+    }
+
+    .little-machine--title {
+      color: $color--black;
+      font-size: 14px;
+    }
+
+    .little-machine--abstract,
+    .little-machine--url {
+      color: $color--dark-grey;
+      font-size: 13px;
+      margin-top: 4px;
+    }
+
+    .little-machine--url {
+      color: $color--blue;
+    }
+  }
+}
+
+
+[data-mozaction="switchtab"] .result--segment[data-mozaction~="switchtab"],
+[data-mozaction="other"] .result--segment[data-mozaction~="other"] {
+  display: inline-block;
+}
+
+.icon {
+  background-repeat: no-repeat;
+  background-size: $size--icon;
+}
+
+.icon--action {
+  [data-mozaction="switchtab"] & {
+    background-image: url(../images/icons/tab.svg);
+  }
+
+  [data-mozaction="switchtab"].result--selected & {
+    background-image: url(../images/icons/tab-selected.svg);
+  }
+}
+
+.icon--recommendation {
+  background-image: url(../images/icons/recommendation.svg);
+
+  .result--selected & {
+    background-image: url(../images/icons/recommendation-selected.svg);
+  }
 }
 
 .icon {
```

Comment:
Cool, I was just playing with [**sass-lint**](https://github.com/sasstools/sass-lint) and noticed a few warnings.

Not sure if it's worth adding to this repo. I wasn't able to get the vanilla version working, but it works fine if I use the **gulp-sass-lint** plugin, although adding Gulp just for that task may be a bit excessive. Currently reports about [53 warnings](https://gist.github.com/pdehaan/89175a6bbcf052c396ce), but some are silly and should be silenced.